### PR TITLE
erts: Get rid of various things scheduled for removal in OTP 27

### PR DESCRIPTION
**Pull request by jhogberg**

### CODE REVIEW

File: erts/emulator/beam/emu/emu_load.c [hunk 1]
```diff
@@ -665,15 +665,15 @@ void beam_load_finalize_code(LoaderState* stp, struct erl_module_instance* inst_
...
                 funp = (ErlFunThing *)fun_val(literal);
-                ASSERT(funp->creator == am_external);
+                ASSERT(funp->external == 1);
 
                 funp->entry.fun = fun_entry;
 
                 funp->next = literal_area->off_heap;
                 literal_area->off_heap = (struct erl_off_heap_header *)funp;
 
                 ASSERT(erts_init_process_id != ERTS_INVALID_PID);
```
Comment (michalmuskala) [inline erts/emulator/beam/emu/emu_load.c:675]: Is this assertion still necessary?

File: erts/emulator/beam/erl_bif_info.c [hunk 2]
```diff
@@ -3674,7 +3674,7 @@ fun_info_2(BIF_ALIST_2)
         hp = HAlloc(p, 3);
         break;
     case am_pid:
-        val = is_local_fun(funp) ? funp->creator : am_undefined;
+        val = is_local_fun(funp) ? erts_init_process_id : am_undefined;
         hp = HAlloc(p, 3);
         break;
```
Comment (michalmuskala) [inline erts/emulator/beam/erl_bif_info.c:3679]: Does it make sense to keep this option around at this point?

File: erts/emulator/beam/emu/emu_load.c (hunk 1 above)
Comment (jhogberg) [reply erts/emulator/beam/emu/emu_load.c:675]: No, good catch. :)

File: erts/emulator/beam/erl_bif_info.c (hunk 2 above)
Comment (jhogberg) [reply erts/emulator/beam/erl_bif_info.c:3679]: No, it's scheduled for removal altogether in OTP 28.

File: erts/emulator/beam/jit/asm_load.c [hunk 3]
```diff
@@ -997,15 +997,15 @@ void beam_load_finalize_code(LoaderState *stp,
...
                 funp = (ErlFunThing *)fun_val(literal);
-                ASSERT(funp->creator == am_external);
+                ASSERT(funp->external == 1);
 
                 funp->entry.fun = fun_entry;
 
                 funp->next = literal_area->off_heap;
                 literal_area->off_heap = (struct erl_off_heap_header *)funp;
 
                 ASSERT(erts_init_process_id != ERTS_INVALID_PID);
```
Comment (michalmuskala) [inline erts/emulator/beam/jit/asm_load.c]: Similar for assert here

**Review by michalmuskala — APPROVED** [review]

**Pull request merged**